overlord/hookstate/hookmgr.go

```diff
@@ -75,9 +66,10 @@ type hookSetup struct {
 func Manager(s *state.State) (*HookManager, error) {
 	runner := state.NewTaskRunner(s)
 	manager := &HookManager{
-		state:      s,
-		runner:     runner,
-		repository: newRepository(),
+		state:          s,
+		runner:         runner,
+		repository:     newRepository(),
+		activeHandlers: newHandlerCollection(),
```

Comment:
s/activeHandlers/handlers/ (we don't have any inactive ones)

I've mentioned this in earlier reviews: it doesn't feel like the additional abstraction there is gaining us much. As a thought experiment, try to inline a map and a simple mutex here, and see how much code you need to write here, vs. how much code you drop.

That said, I believe what we really want here is a map of context, not handlers. The reason being that context is a concrete and common implementation, and it would be simple and convenient to introduce a `HookHandler` method on context to navigate from a context into its handler. Doing that in the opposite direction is boring, because every single handler will need to reimplement the same method to return its context.